File: Level-2/45.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int area = brown + yellow;
        
        for(int i = 1; i <= area; i++) {
            int row = i; // 세로
            int col = area / i; // 가로
            
            if(col >= row) {
                if((col - 2) * (row - 2) == yellow) {
                    answer[0] = col;
                    answer[1] = row;
                    break;
                }
            }else {
                break;
            }
        }
        
        return answer;
    }
}

File: Level-2/45.js
function solution(brown, yellow) {
    for(let i = 1; i <= Math.sqrt(yellow); i++) {
        if(yellow % i === 0) {
            const a = i + 2;
            const b = (yellow / i) + 2;
            if(a * b - yellow === brown) {
                return [b, a];
            }
        }
    }
}
